tests/testthat/test-read_target_file.R

```diff
@@ -0,0 +1,303 @@
+  skip_if_offline()
+  # read in oracle-output file with hub schema
+  file <- read_target_file("oracle-output.csv", hub_path)
+  expect_s3_class(file, "tbl_df")
+  expect_equal(
+    names(file),
+    c(
+      "location", "target_end_date", "target", "output_type", "output_type_id",
+      "oracle_value"
+    )
+  )
+  expect_true(all(is.na(file$output_type_id[file$output_type == "quantile"])))
+  expect_equal(dim(file), c(200340L, 6L))
+  expect_equal(
+    purrr::map_chr(file, ~ class(.x)),
+    c(
+      location = "character", target_end_date = "Date", target = "character",
+      output_type = "character", output_type_id = "character", oracle_value = "numeric"
+    )
+  )
+  expect_equal(
+    purrr::map_chr(file, ~ typeof(.x)),
+    c(
+      location = "character", target_end_date = "double", target = "character",
+      output_type = "character", output_type_id = "character", oracle_value = "double"
+    )
+  )
+
+  # read in oracle-output file as character columns
+  chr_file <- read_target_file("oracle-output.csv", hub_path,
+    coerce_types = "chr"
+  )
+  expect_s3_class(chr_file, "tbl_df")
+  expect_equal(dim(chr_file), c(200340L, 6L))
+  expect_true(all(is.na(file$output_type_id[file$output_type == "quantile"])))
+  expect_equal(
+    purrr::map_chr(chr_file, ~ class(.x)),
+    c(
+      location = "character", target_end_date = "character", target = "character",
+      output_type = "character", output_type_id = "character", oracle_value = "character"
+    )
+  )
+})
+
+test_that("read_target_file errors correctly.", {
+  skip_if_offline()
+  expect_error(
+    read_target_file("non-existent.csv", hub_path),
+    regexp = "Could not determine target type of file"
+  )
+  expect_error(
+    read_target_file(c(
+      "time-series.csv",
+      "oracle-output.csv"
+    ), hub_path),
+    regexp = "Assertion on 'target_file_path' failed: Must have length 1, but has length 2."
+  )
+  expect_error(
+    read_target_file("time-series", hub_path),
+    regexp = "No file exists at path"
+  )
+})
+
+test_that("read_target_file works with split timeseries file.", {
+  skip_if_offline()
+
+  fs::dir_copy(hub_path, dir_hub_path)
+  path <- abs_file_path("time-series.csv", dir_hub_path, subdir = "target-data")
+  # Read timeseries data from single file
+  dat <- arrow::read_csv_arrow(path)
+  # Delete single time-series file in preparation for creating time-series directory
+  fs::file_delete(path)
+
+  # Create a seperate file for each target in a time-series directory
+  dir <- fs::path(dir_hub_path, "target-data", "time-series")
+  fs::dir_create(dir)
+  split(dat, dat$target) |> purrr::iwalk(
+    ~ {
+      target <- gsub(" ", "_", .y)
+      path <- file.path(dir, paste0("target-", target, ".csv"))
+      arrow::write_csv_arrow(.x, file = path)
+    }
+  )
+  fs::dir_ls(dir)
+
+  # read in one of the time-series file with hub schema
+  file <- read_target_file("time-series/target-wk_inc_flu_hosp.csv", dir_hub_path)
```

Comment:
I like this because it provides an avenue for hubs that do not want to combine targets into a single time series file.